src/views/players/view_players.cpp

```diff
@@ -100,44 +100,65 @@ namespace big
 
 		if (!*g_pointers->m_gta.m_is_session_started && player_count < 2)
 			return;
+
+		const auto style = ImGui::GetStyle();
 		float window_pos = 110.f + g_gui_service->nav_ctr * ImGui::CalcTextSize("W").y
-		    + g_gui_service->nav_ctr * ImGui::GetStyle().ItemSpacing.y
-		    + g_gui_service->nav_ctr * ImGui::GetStyle().ItemInnerSpacing.y + ImGui::GetStyle().WindowPadding.y;
+		    + g_gui_service->nav_ctr * style.ItemSpacing.y + g_gui_service->nav_ctr * style.ItemInnerSpacing.y
+		    + style.WindowPadding.y;
 
 		ImGui::SetNextWindowSize({300.f, 0.f});
 		ImGui::SetNextWindowPos({10.f, window_pos});
 		ImGui::PushStyleVar(ImGuiStyleVar_WindowPadding, {2.0f, 2.0f});
 
 		if (ImGui::Begin("playerlist", nullptr, window_flags))
 		{
-			const auto style = ImGui::GetStyle();
-			float window_height = (
-				ImGui::CalcTextSize("A").y + style.FramePadding.y * 2.0f + style.ItemSpacing.y) // button size
-				* player_count // amount of players
-				+ (player_count > 1) * ((style.ItemSpacing.y * 2) + 1.f) // account for ImGui::Separator spacing
-				+ (player_count == 1) * 2.f; // some arbitrary height to make it fit
-			// used to account for scrollbar width
-			has_scrollbar = window_height + window_pos > (float)*g_pointers->m_gta.m_resolution_y - 10.f;
-
-			// basically whichever is smaller, the max available screenspace or the calculated window_height
-			window_height = has_scrollbar ? (float)*g_pointers->m_gta.m_resolution_y - (window_pos + 40.f) : window_height;
-
 			ImGui::PushStyleColor(ImGuiCol_FrameBg, {0.f, 0.f, 0.f, 0.f});
 			ImGui::PushStyleColor(ImGuiCol_ScrollbarBg, {0.f, 0.f, 0.f, 0.f});
 
-			if (ImGui::BeginListBox("##players", {ImGui::GetWindowSize().x - ImGui::GetStyle().WindowPadding.x * 2, window_height}))
+			const auto style  = ImGui::GetStyle();
+			auto window_width = ImGui::GetWindowSize().x - style.WindowPadding.x * 2;
+
+			// render search
+			static std::string search_player_name;
+			ImGui::SetNextItemWidth(window_width);
+			if (components::input_text_with_hint("###search_player_name", "SEARCH"_T, search_player_name))
+				std::transform(search_player_name.begin(), search_player_name.end(), search_player_name.begin(), ::tolower);
+			auto search_height = ImGui::GetItemRectSize().y; // Get the size of the last element
+			//
+
+			float raw_height = (ImGui::CalcTextSize("A").y + style.FramePadding.y * 2.0f + style.ItemSpacing.y) // button size
```

Comment:
ImGui::CalcTextSize("A").y + style.FramePadding.y * 2.0f + style.ItemSpacing.y;

This could be factored out into a static const since it is not likely to change at runtime after calculation.

```
static const auto button_size = ImGui::CalcTextSize("A").y + style.FramePadding.y * 2.0f + style.ItemSpacing.y;
const auto raw_height = button_size * player_count + (player_count > 1) * ((style.ItemSpacing.y * 2) + 1.f) + (player_count == 1) * 2.f;
has_scrollbar = raw_height + window_pos > (float)*g_pointers->m_gta.m_resolution_y - 10.f;
const auto window_height = has_scrollbar ? (float)*g_pointers->m_gta.m_resolution_y - (window_pos + search_height + 40.f) : raw_height;
```

Not sure if it would actually speed up the calculations or not, but you also want to remember, the C++ compiler enforces PEMDAS. Splitting out things that need to be calculated before hand is the easiest way to tell the compiler which can be optimized. 

It also makes it more readable instead of one giant-ass block of code.

Reply:
i am gonna look into this

Reply:
I believe -

`m_resolution_y`  can change, `window_pos` when you popup different menus in nav menu. `ItemSpacing, CalcTextSize, FramePadding` and infact everything changes when you rescale UI.

Reply:
we can remove certain code by assuming fix height of container that is (float)*g_pointers->m_gta.m_resolution_y - (window_pos + search_height + 40.f) and width related code need to be checked as well.

Reply:
also `window_pos` can also be determined simply if  `view::heading()`, `view::navigation()` & `view::players()` are part of same group.

Reply:
i thought of doing this what you said and i did something. It is true one can calculate the size and determine it again when UI is rescaled. I will share the changes.

Reply:
https://github.com/lonelybud/lonelyMenu/commit/f2c062541b36417c05a09893dbb022266fb4094a#diff-cea3cc35a3bc3e5a06c30d7eedb274e4e564ddc8ee39997a221e515d19931493

the code is weird in the sense that it has some code quality and dependency issues like calling `update_gui_info` whenever we need to update the gui variables. So it will update the gui when menu is toggled, players join/leaves/reset/click, navigation change. Not a desired solution but still you can find some part interesting,